src/client/oicp/CpoOICPClient.ts

```diff
@@ -0,0 +1,1272 @@
+    payload.ActionType = actionType;
+    payload.PricingProductData = pricingProductData;
+
+    // Log
+    Logging.logDebug({
+      tenantID: this.tenant.id,
+      action: ServerAction.OICP_PUSH_PRICING_PRODUCT_DATA,
+      message: `Push pricing product data from tenant: ${this.tenant.id}`,
+      module: MODULE_NAME, method: 'pushPricingProductData',
+      detailedMessages: { payload }
+    });
+
+    // Call Hubject
+    await this.axiosInstance.post(fullUrl, payload
+    ).then(function(response) {
+      console.log('Success! pushPricingProductData: ', response);
+      pushPricingProductDataResponse = response.data as OICPAcknowledgment;
+    }).catch((error) => {
+      console.log('Error! pushPricingProductData: ', error.response);
+    });
+    return pushPricingProductDataResponse;
+  }
+
+  /**
+   * Send Charging Notification Start
+   */
+  async sendChargingNotificationStart(transaction: Transaction): Promise<OICPAcknowledgment> {
+    let notificationStartResponse: OICPAcknowledgment = {} as OICPAcknowledgment;
+    // Check for input parameter
+    if (!transaction.oicpData) {
+      throw new BackendError({
+        source: transaction.chargeBoxID,
+        action: ServerAction.OICP_SEND_CHARGING_NOTIFICATION_START,
+        message: `OICP data does not exists on Session ID '${transaction.id}'`,
+        module: MODULE_NAME, method: 'sendChargingNotificationStart',
+      });
+    }
+    if (!transaction.oicpData.session) {
+      throw new BackendError({
+        source: transaction.chargeBoxID,
+        action: ServerAction.OICP_SEND_CHARGING_NOTIFICATION_START,
+        message: `OICP Session data does not exists on Session ID '${transaction.id}'`,
+        module: MODULE_NAME, method: 'sendChargingNotificationStart',
+      });
+    }
+
+    // Get notification endpoint url
+    // for QA - environment: https://service-qa.hubject.com/api/oicp/notificationmgmt/v11/charging-notifications
+    // for PROD - environment: https://service.hubject.com/api/oicp/notificationmgmt/v11/charging-notifications
+    const fullUrl = this.getEndpointUrl('notifications', ServerAction.OICP_SEND_CHARGING_NOTIFICATION_START);
+
+    // Build payload
+    const payload: OICPChargingNotificationStartCpoSend = {} as OICPChargingNotificationStartCpoSend;
+    payload.Type = OICPChargingNotification.Start;
+    payload.SessionID = transaction.oicpData.session.id;
+    if (transaction.id) {
+      payload.CPOPartnerSessionID = String(transaction.id); // Optional
+    }
+    payload.EMPPartnerSessionID = transaction.oicpData.session.empPartnerSessionID; // Optional
+    payload.Identification = transaction.oicpData.session.identification; // Optional
+    payload.EvseID = transaction.oicpData.session.evse.EvseID;
+    payload.ChargingStart = transaction.timestamp;
+    payload.SessionStart = transaction.oicpData.session.start_datetime; // Optional
+    payload.MeterValueStart = parseFloat((transaction.meterStart / 1000).toFixed(3)); // Optional
```

Comment:
Always user our helper methods when available for type conversion. For all types conversion.

Reply:
done